tests/test_edgeql_functions.py

```diff
@@ -2254,9 +2267,21 @@ class TestEdgeQLFunctions(tb.QueryTestCase):
             {0.01},
         ])
 
+    @unittest.skip('''
+        The cast distance of int16 is 2 for mean(int64) and
+        mean(float64). Currently instead of an error one of the
+        implementaitons will be used at random, so the test failure is
+        unstable.
+
+        Once the cast distance ambiguity is fixed the next issue will
+        be making sure that only decimals produce decimal result,
+        everything else is float32 or float64.
+    ''')
```

Comment:
Just fix this -- `compiler/func.py` lines 96..110.

Reply:
Currently, a "fix" for ambiguity will result in breaking many tests (such as `ln(<int16>10)` being unable to choose between `int64` and `float64` implementations). Fixing that, will backfire on how `float32` + `int16` works, namely it will result in a `float64` or it will require explicit versions for every numeric scalar of every std and math function. We also don't yet have validation that the SQL function body returns what is promised in EdgeQL. All of this sounds like a whole separate PR that's concentrating on *functions*.

The bottom line is that this fix is not in the scope of this PR, but defining sane *implicit casts* and *operators* is. So it seems that much like with `decimal` we need to adopt a specific doctrine w.r.t. `float32`. Just how special is it? How contagious? And why? Similarly, how special are `int16` and `int32`?